### DISCUSSION TITLE
How to import the models and predict on a fresh dataset?

### DISCUSSION THREAD

**Discussion by thepraveen19** [safe-graph/GNN-FakeNews#7] [Q&A] [answered]
### Discussed in https://github.com/safe-graph/GNN-FakeNews/discussions/5

<div type='discussions-op-text'>

<sup>Originally posted by **thepraveen19** December 24, 2021</sup>
I am working on a twitter dataset where I want to classify the tweets as real or fake. Is there a way to import the model and use it for predictions?

Please help!</div>

**Comment by YingtongDou** [comment, marked as answer]
Hi, to use the GNN models in this repo to classify a tweet, you need to build a tree-structured tweet propagation graph based on the retweets of the given tweets, which you can refer to the [paper](https://arxiv.org/pdf/2104.12259.pdf) for more instructions.

Once build the graph, you can use the pretrained model on the UPFD dataset to classify your tweets but I strongly suggest curating your own training data since there might be domain shifts for different news sets.
